File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> complement = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int comp = target - nums[i];
            if (complement.containsKey(comp)) {
                int[] res = {complement.get(comp), i};
                return res;
            }
            complement.put(nums[i], i);
        }
        return null;
    }
}

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-1.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        if sum(nums) % 2 != 0:
            return False
        target = sum(nums) / 2

        def sub(nums, target):
            if target == 0:
                return True
            if nums == [] or target < 0:
                return False
            return sub(nums[1:], target - nums[0]) or sub(nums[1:], target)
        
        return sub(nums, target)